core/Symbols.h

```diff
@@ -617,12 +617,31 @@ class ClassOrModule final {
         return members_;
     };
 
+    struct MembersStableOrder {
+        std::vector<std::pair<NameRef, SymbolRef>> members;
+
+        template <typename P> void accumulateMembers(const GlobalState &gs, const ClassOrModule &klass, P predicate) {
```

Comment:
I like the concept but maybe we can tweak the name + add some docs?

Name suggestion:

- `accumuateSortedFilteredMembers`
- ...?

Ideally a doc comment would explain more about when to use this and how to think about the fact that it has persistent state (e.g., you're expected to look in the `members` as an out parameter, almost like the return of this function, though see elsewhere for my question of whether that API is what we really want).